app/features/syllabus_generator/tools.py

```diff
@@ -0,0 +1,347 @@
+from langchain_core.pydantic_v1 import BaseModel, Field
+from typing import List
+from app.services.logger import setup_logger
+
+from langchain_core.output_parsers import JsonOutputParser
+from langchain_google_genai import GoogleGenerativeAI
+from langchain_core.prompts import PromptTemplate
+from app.features.syllabus_generator.document_loaders import read_text_file
+
+from fastapi import HTTPException
+
+
+logger = setup_logger(__name__)
+
+class SyllabusRequestArgs:
+    def __init__(self, 
+                 grade_level: str, 
```

Comment:
Look into consolidating into one variable whose interface is the parameter for the function

Reply:
Done. We decoupled the SyllabusGeneratorArgs with a Pydantic schema whose interface is the parameter for the function.